### PR TITLE
Add make image

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,10 +48,8 @@ run:
 clean:
 	@rm -rf $(TARGET_DIR)
 
-# image: imagebuilder
-# 	@if [ $${USE_IMAGE_STREAM:-false} = false ] && [ $${SKIP_BUILD:-false} = false ] ; \
-# 	then $(IMAGE_BUILDER) -t $(IMAGE_TAG) . $(IMAGE_BUILDER_OPTS) ; \
-# 	fi
+image: $(info Building Docker image...)
+	@docker build -f $(DOCKERFILE) -t $(IMAGE_TAG) $(CURPATH) $(IMAGE_BUILD_OPTS)
```
Comment (pliurh) [inline Makefile]: I think we may give develops choice here, so they can chose to use which tool to build the image, such as podman, buildah or docker.

File: Makefile [hunk 2]
```diff
@@ -6,23 +6,22 @@ GOLIST=go list
...
-export IMAGE_TAG_CMD?=docker tag
+IMAGE_BUILD_OPTS=${MAGE_BUILDER_OPTS:- ""}
+DOCKERFILE=${DOCKERFILE:- "Dockerfile"}
+DOCKERFILE=Dockerfile
 
 export APP_NAME=sriov-network-operator
 APP_REPO=github.com/openshift/$(APP_NAME)
 TARGET=$(TARGET_DIR)/bin/$(APP_NAME)
-export IMAGE_TAG=quay.io/pliurh/$(APP_NAME):latest
+export IMAGE_TAG=quay.io/openshift/$(APP_NAME):latest
```
Comment (pliurh) [inline Makefile]: The image in openshift repo is built by openshift CI system. Building image with Makefile shall only be use for test purpose. So I think we shall use different tag for the test image. Maybe we can use an environment variable here.

File: Makefile [hunk 3]
```diff
@@ -6,23 +6,22 @@ GOLIST=go list
 GOFMT=gofmt
 BUILD_GOPATH=$(TARGET_DIR):$(TARGET_DIR)/vendor:$(CURPATH)/cmd
 
-IMAGE_BUILDER_OPTS=
-IMAGE_BUILDER?=imagebuilder
-IMAGE_BUILD=$(IMAGE_BUILDER)
-export IMAGE_TAG_CMD?=docker tag
+IMAGE_BUILD_OPTS=${MAGE_BUILDER_OPTS:- ""}
```
Comment (zshi-redhat) [inline Makefile]: `MAGE_BUILDER_OPTS` -> `IMAGE_BUILDER_OPTS`

File: Makefile [hunk 4]
```diff
@@ -6,23 +6,22 @@ GOLIST=go list
...
+DOCKERFILE=${DOCKERFILE:- "Dockerfile"}
+DOCKERFILE=Dockerfile
```
Comment (zshi-redhat) [inline Makefile]: Should this be removed? `DOCKERFILE` already be assigned in above cmd.

**Comment by Mmduh-483** [comment]
@pliurh @zshi-redhat I fixed the comments above, can u please re-check it

File: Makefile [hunk 5]
```diff
@@ -6,23 +6,22 @@ GOLIST=go list
...
-export IMAGE_TAG_CMD?=docker tag
+IMAGE_BUILDER?=@docker
+IMAGE_BUILD_OPTS?=
+DOCKERFILE?=Dockerfile
 
 export APP_NAME=sriov-network-operator
 APP_REPO=github.com/openshift/$(APP_NAME)
 TARGET=$(TARGET_DIR)/bin/$(APP_NAME)
-export IMAGE_TAG=quay.io/pliurh/$(APP_NAME):latest
+IMAGE_TAG?=quay.io/openshift/$(APP_NAME):latest
```
Comment (zshi-redhat) [inline Makefile]: OpenShift uses `Dockerfile.rhel7` for image building, we might want to name this tag as a community version.
For example, to make this consistent with other sriov components like device plugin or cni, use `nfvpe` prefix and setup travis to upload image upon any repo changes.

Comment (pliurh) [reply Makefile]: +1, or we can just use a environment variable here, so that developers can chose whatever repo name they want to use.

Comment (Mmduh-483) [reply Makefile]: I am working on another PR for travis job
@zshi-redhat @pliurh do you have other comments?

**Pull request merged**